Sources/Hyperspace/HTTP/HTTP.swift

```diff
@@ -100,31 +108,52 @@ public struct HTTP {
     public struct Body: Equatable {
         
         /// The raw body data to be attached to the HTTP request
-        public let data: Data
+        public let data: Data?
+        public let additionalHeaders: [HeaderKey: HeaderValue]
```

Comment:
Seeing how this is used, I'm wondering if we should abstract the concept up to a `contentType` property? I suppose this approach is technically more flexible, but I would imagine you would usually want to assign as many of your headers directly on the `Request` itself. Thoughts?
```swift
public let contentType: HeaderValue?
```

Reply:
That could work as well, for sure. I'm trying to think of any times I've wanted to add 2 header key/value pairs when I've set a particular HTTP body and coming up empty (not sure if any come to anyone else's mind).

It could also be another convenience initializer?

```swift
init(data: Data?, contentType: HeaderValue?) {
 self.init(data: data, additionalHeaders: contentType.map { [.contentType: $0] })
}
```

Reply:
That could work, though now that I'm thinking about it, we might already have what we need with the static `json(_:encoder:additionalHeaders:)` and `urlForm(using:additionalHeaders:)` functions that you created below. I think we'll pretty much always be using those when we need to construct a `HTTP.Body`, right?

Reply:
Yea, the main reason I only left it defaulted is there are some services that have some weirdness associated with it -
for instance Airship uses a custom `application/vnd.urbanairship+json; version=3` content type for JSON. I can't personally think of anywhere that would need 2 headers, but I'm almost positive there's someone out there 😄